coral-spark/src/main/java/com/linkedin/coral/spark/transformers/CastToNamedStructTransformer.java

```diff
@@ -0,0 +1,53 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.spark.transformers;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlIdentifier;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlRowTypeNameSpec;
+import org.apache.calcite.sql.SqlRowTypeSpec;
+import org.apache.calcite.sql.parser.SqlParserPos;
+
+import com.linkedin.coral.common.transformers.SqlCallTransformer;
+import com.linkedin.coral.hive.hive2rel.functions.HiveNamedStructFunction;
+
+
+/**
+ * This transformer transforms `CAST(ROW: RECORD_TYPE)` to `named_struct` in Spark.
```

Comment:
Not sure if it's suitable to call it `Coral IR` function, I think it's a function in CoralSqlNode converted from Coral IR?

Reply:
in the path:
`source SQL -> sourceSqlNode1 -> CoralSqlNode1 -> CoralRelNode -> CoralSqlNode2 -> TargetLangSqlNode -> target SQL`

if an operator is present in CoralSqlNode1, CoralRelNode, CoralSqlNode2 and we transform this operator when we go to LanguageSqlNode, such operators can be called Coral IR functions.

Reply:
as per the current status of our code base - 
`hive SQL -> HiveSqlNode (same as CoralSqlNode1) -> CoralRelNode -> CoralSqlNode2 -> SparkSqlNode -> Spark SQL`
I think this operator qualifies as Coral Operator

Reply:
Thanks for clarification, do you mean an operator is a Coral IR function if it appears in `CoralSqlNode1 / CoralRelNode / CoralSqlNode2` or `CoralSqlNode1 & CoralRelNode & CoralSqlNode2`?
Not sure because it doesn't appear in `CoralSqlNode1`.

Reply:
for a corresponding input hive SQL, what is the operator we use in the SqlNode representation? how do we represent the named_struct on LHS?

but generally yes, an operator present in CoralSqlNode1 would be present in CoralRelNode and then generated back in CoralSqlNode2.

Reply:
It's still represented as `named_struct` in CoralSqlNode1, and Calcite converts it to `CAST(ROW: RECORD_TYPE)` in RelNode.

Reply:
I see. In that case, could you investigate if it's possible to add a lightweight override in `CoralRelNodeToCoralSqlNode` to such that the coralSqlNode2 generated uses the same operator `named_struct` as coralSqlNode1?

Reply:
Opened #357 for tracking.